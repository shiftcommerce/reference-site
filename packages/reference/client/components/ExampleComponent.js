import { Component } from 'react'

class ExampleComponent extends Component {
  render () {
    const {
      thisProp,
      thatProp
    } = this.props

    return (
      <div>
        `The output of { thisProp } + { thatProp } is { thisProp + thatProp }`
      </div>
    )
  }
}

export default ExampleComponent
