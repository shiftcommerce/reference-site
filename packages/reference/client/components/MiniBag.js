// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Link from 'next/link'
import classNames from 'classnames'
import Cookies from 'js-cookie'

// Actions
import { readCart } from '../actions/cartActions'

// Objects
import Button from '../objects/Button'

export class MiniBag extends Component {
  componentDidMount () {
    this.props.dispatch(readCart())
  }

  renderViewYourBagLink (lineItemCount) {
    return (
      <Link href='/cart'>
        <div className='c-minibag__cart'>
          <a aria-label='View your cart'>
            <span className='c-minibag__cart-count' >
              { lineItemCount }
            </span>
          </a>
          <div className='c-minibag__cart-label--small'>
            <a>
              Basket
            </a>
          </div>
          <div className='c-minibag__cart-label--large'>
            <a>
              View Your Bag
            </a>
          </div>
        </div>
      </Link>
    )
  }

  renderCheckoutButton (lineItemCount) {
    const signedIn = Cookies.get('signedIn')

    return (
      <div className='c-minibag__checkout'>
        <Link href={(lineItemCount > 0) ? '/checkout' : ''}>
          <Button className={classNames({ 'o-button--disabled': (lineItemCount === 0) })} label="Checkout" status="primary" aria-label='Go to checkout'>
          </Button>
        </Link>
        { this.renderMyAccount(signedIn) }
        { this.renderLogout(signedIn) }
      </div>
    )
  }

  renderMyAccount (signedIn) {
    return (
      <Link href={(signedIn) ? '/account/myaccount' : '/account/login'} as='/account/myaccount' >
        <a className='o-header__myaccount'>My Account</a>
      </Link>
    )
  }

  renderLogout (signedIn) {
    if (signedIn) {
      return (
        <a href='/account/logout' className='o-header__myaccount'>
          Logout
        </a>
      )
    }
  }

  render () {
    const { cart } = this.props
    const lineItemCount = cart.totalQuantity

    return (
      <div className='c-header__minibag'>
        { this.renderViewYourBagLink(lineItemCount) }
        { this.renderCheckoutButton(lineItemCount) }
      </div>
    )
  }
}

function mapStateToProps (state) {
  const { cart } = state
  return { cart }
}

export default connect(mapStateToProps)(MiniBag)
