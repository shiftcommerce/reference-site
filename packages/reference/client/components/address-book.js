// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'

// Action creators
import { deleteAddressBookEntry, setAddress } from '../actions/address-book-actions'
import { editForm } from '../actions/checkout-actions'

class AddressBook extends Component {
  setAddress (address) {
    const { formName, dispatch } = this.props

    dispatch(setAddress(address, formName))
  }

  handleAddressDeleted (address) {
    return this.props.dispatch(deleteAddressBookEntry(address))
  }

  createNewAddress () {
    const { formName, dispatch, onToggleCollapsed } = this.props
    onToggleCollapsed('edit', formName)
    dispatch(editForm(formName))
  }

  renderOptions (address) {
    const { id } = this.props
    const optionLabel = <a><b>{ address.meta_attributes.label.value }</b> -&nbsp; { address.first_name } { address.last_name },&nbsp;
      { address.address_line_1 },&nbsp; { address.postcode },&nbsp; { address.city },&nbsp; { address.country }
    </a>

    return (
      <div className='c-address-book__radio-container'>
        <input type='radio' name='address-book-radio' className='c-address-book__radio' onChange={() => { this.setAddress(address) }} checked={address.id === id} />
        <label htmlFor={address.id}>{ optionLabel }</label>
        <a className='c-address-book__delete-address' label='Delete' onClick={() => this.handleAddressDeleted(address)}>Delete</a>
      </div>
    )
  }

  renderEntries () {
    // Sort array of objects and return a new array
    // with the preferred address as the first object.
    // We always want to display the preferred-address as the first option
    const entries = this.props.addressBook.sort(function (a, b) {
      return b.preferred_shipping - a.preferred_shipping
    })

    return (
      <div className='c-address-book__entries'>
        { entries.map(address => {
          return (
            <div key={address.id}>
              { this.renderOptions(address) }
            </div>
          )
        }) }
      </div>
    )
  }

  render () {
    const { formName } = this.props

    return (
      <div className='c-address-book'>
        <h3>Your addresses</h3>
        { this.renderEntries() }
        <a className='c-address-book__new-address' onClick={() => this.createNewAddress(formName)}>+ Add a new address</a>
      </div>
    )
  }
}

const mapStateToProps = ({ checkout: { addressBook, shippingAddress: { id } } }) => {
  return { addressBook, id }
}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: dispatch
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddressBook)
