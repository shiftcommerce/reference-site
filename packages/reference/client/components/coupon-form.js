// Libraries
import { Component } from 'react'
import { Formik, Form, Field, ErrorMessage } from 'formik'
import { connect } from 'react-redux'

// Actions
import { readCart } from '../actions/cart-actions'

// Request
import { addCartCouponRequest } from '../requests/cart-requests'

// Lib
import ApiClient from '../lib/api-client'

// Objects
import Button from '../objects/button'

export class CouponForm extends Component {
  async submitCoupon (couponCode) {
    const request = addCartCouponRequest(couponCode)
    const response = await new ApiClient().post(request.endpoint, request.body)
    if (response.status === 201) {
      return response.data
    }
    throw response.data
  }

  constructor () {
    super()
    this.handleSubmit = this.handleSubmit.bind(this)
    this.updateCart = this.updateCart.bind(this)
  }

  handleSubmit (values, { setSubmitting, setErrors }) {
    return this.submitCoupon(values.couponCode)
      .then(this.updateCart)
      .catch((error) => this.setAPIError(error, setErrors))
      .finally(() => setSubmitting(false))
  }

  updateCart () {
    this.props.dispatch(readCart({ force: true }))
  }

  setAPIError (error, setErrors) {
    setErrors({ couponCode: error.errors[0]['title'] })
  }

  validate (values) {
    let errors = {}
    if (!values.couponCode) errors.couponCode = 'Please enter a coupon code'
    return errors
  }

  renderForm () {
    return (
      <Formik
        initialValues={{ couponCode: '' }}
        validate={this.validate}
        onSubmit={this.handleSubmit}
      >
        { () => (
          <Form>
            <div className='c-coupon-form__field'>
              <Field type='text' name='couponCode' className='c-coupon-form__input' placeholder='Enter Promotion Code' />
              <Button
                aria-label='Apply Gift Code'
                containerClassName='c-coupon-form__button-container'
                className='c-coupon-form__button o-button--sml'
                label='Apply'
                status='primary' />
            </div>
            <ErrorMessage name='couponCode' component='div' className='c-coupon-form__error' />
          </Form>
        ) }
      </Formik>
    )
  }

  render () {
    return (
      <div className='c-coupon-form'>
        <div aria-label='Use gift card or rewards code' className='c-gift__wrapper'>
          <p className='c-coupon-form__title'>Promotion Code</p>
          { this.renderForm() }
        </div>
      </div>
    )
  }
}

export default connect()(CouponForm)
