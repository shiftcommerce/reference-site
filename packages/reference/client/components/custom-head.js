import Head from 'next/head'
import { Component } from 'react'

class CustomHead extends Component {
  render () {
    return <Head>
      <meta name='viewport' content='width=device-width, initial-scale=1' />
      <script src='https://js.stripe.com/v3/' />
      <link rel='icon' type='image/png' sizes='32x32' href='/static/favicon.png' />
    </Head>
  }
}

export default CustomHead
