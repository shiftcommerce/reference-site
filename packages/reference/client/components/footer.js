// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'

// Objects
import { Image } from 'shift-react-components'

export class Footer extends Component {
  renderFooter () {
    return (
      <div className='c-footer'>
        <div className='c-footer__content'>
          <div className='c-footer__links'>
            <a>FAQs</a><a>About</a><a>Delivery</a><a>Stores</a><a>Contact</a>
          </div>
          <div className='c-footer__social'>
            <div className='c-footer__social-text'>
              <a>Connect with ShopGo</a>
            </div>
            <div className='c-footer__social-links'>
              <Image src='/static/facebook.svg' />
              <Image src='/static/instagram.svg' />
              <Image src='/static/twitter.svg' />
            </div>
          </div>
        </div>
      </div>
    )
  }

  render () {
    return (
      <div>
        { this.renderFooter() }
      </div>
    )
  }
}

export default connect()(Footer)
