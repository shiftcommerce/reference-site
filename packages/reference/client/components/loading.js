import React from 'react'
import ReactLoading from 'react-loading'

const Loading = ({ type, color }) => (
  <ReactLoading
    type={'spin'}
    color={'#f5c6cb'}
    height={50}
    width={50}
    delay={250}
    className={'c-loading u-padding-top-xl'}
  />
)

export default Loading
