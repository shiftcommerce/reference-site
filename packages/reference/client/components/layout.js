// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'
import Link from 'next/link'
import classNames from 'classnames'

// Objects
import { Footer, Image, Logo, NavBar, SearchBar } from 'shift-react-components'

// Components
import MiniBag from './mini-bag'
import CustomHead from './custom-head'

// Stylesheet
import '../scss/main.scss'

export class Layout extends Component {
  // serviceWorker () {
  //  // Install service worker only in production environment
  //   if (process.env.NODE_ENV === 'production') {
  //    // Registration of service worker
  //     if (typeof navigator !== 'undefined' && 'serviceWorker' in navigator) {
  //       navigator.serviceWorker.getRegistration('/app').then(registration => {
  //        // Check if service worker has already registered
  //        // register only if not yet done
  //         if (registration === undefined) {
  //           navigator.serviceWorker.register('/serviceWorker.js', { scope: '/' }).then(registration => {
  //            // Successfully registered the Service Worker
  //             console.log('Service Worker registration successful with scope: ', registration.scope)
  //           }).catch(err => {
  //            // Failed to register the Service Worker
  //             console.log('Service Worker registration failed: ', err)
  //           })
  //         }
  //       })
  //     } else {
  //       console.log('Service workers are not supported.')
  //     }
  //   }
  // }
  constructor (props) {
    super(props)
    this.state = {
      shrunk: false,
      toggleShowClass: false
    }

    this.handleScroll = this.handleScroll.bind(this)
    this.toggleDropDown = this.toggleDropDown.bind(this)
  }

  componentDidMount () {
    window.addEventListener('scroll', this.handleScroll)
  }

  componentWillUnmount () {
    window.removeEventListener('scroll', this.handleScroll)
  }

  handleScroll () {
    const scrollTop = document.documentElement.scrollTop || document.body.scrollTop
    if (scrollTop > 40 && !this.state.shrunk) {
      this.setState(state => {
        return { shrunk: true }
      })
    } else if (scrollTop <= 40 && this.state.shrunk) {
      this.setState(state => {
        return { shrunk: false }
      })
    }
  }

  renderNav () {
    return (
      <span className='o-nav u-visible-d'>
        <NavBar menu={this.props.menu} />
      </span>
    )
  }

  renderMobileNav () {
    return (
      <span className='o-nav u-hidden-d'>
        <NavBar menu={this.props.menu} />
      </span>
    )
  }

  toggleDropDown () {
    const toggleShow = this.state.toggleShowClass
    this.setState({ toggleShowClass: !toggleShow })
  }

  renderHeaderAccount (loggedIn) {
    const signedIn = loggedIn ? 'My Account' : <Link href='/account/login'><a>Sign In</a></Link>

    return (
      <div className='c-header__account' onClick={ this.toggleDropDown } >
        <Image className='c-header__account-image' src='/static/account-icon.svg' />
        { signedIn }
        { this.renderAccountDropDown() }
      </div>
    )
  }

  renderAccountDropDown () {
    const showClass = this.state.toggleShowClass
    const { loggedIn } = this.props

    const addShowClass = showClass ? 'show' : ''

    if (loggedIn) {
      return (
        <div className={classNames('c-header__dropdown-wrapper', addShowClass)} >
          <div className={classNames('c-header__account-dropdown', addShowClass)} >
            <div className='c-header__callout' />
            <Link href='/account/myaccount'>Order History</Link>
            <Link href='/account/forgotpassword'>Change Password</Link>
            <Link href='/account/logout'>Sign Out</Link>
          </div>
        </div>
      )
    } else {
      return null
    }
  }

  renderHeader () {
    const { loggedIn } = this.props
    const logoSrc = '../static/shopgo-logo.svg'

    if (typeof window === 'undefined' || window.location.pathname !== '/checkout') {
      const headerClasses = classNames('o-header', {
        'o-header--shrunk': this.state.shrunk
      })

      return (
        <>
          <CustomHead />
          <div className={ headerClasses }>
            <div className='o-header__top'>
              <div className='o-header__top-wrapper'>
                <Logo className='o-header__logo' logoSrc={logoSrc} />
                { this.renderMobileNav() }
                { this.renderHeaderAccount(loggedIn) }
                <MiniBag />
                { this.renderSearch() }
              </div>
            </div>
            { this.renderNav() }
          </div>
        </>
      )
    }
  }

  renderSearch () {
    return (
      <span className='c-header__search'>
        <SearchBar query={this.props.query} />
      </span>
    )
  }

  render () {
    const notCheckout = /^(?!\/checkout).*$/
    const bodyClassApplied = (typeof window === 'undefined' || notCheckout.test(window.location.pathname))

    const bodyClasses = classNames({
      'o-body': bodyClassApplied
    })

    return (
      <>
        { this.renderHeader() }
        <div className={bodyClasses}>
          { this.props.children }
        </div>
        <div className='o-footer'>
          <Footer />
        </div>
      </>
    )
  }
}

function mapStateToProps (state) {
  const { login, menu } = state

  return { login, menu }
}

export default connect(mapStateToProps)(Layout)
