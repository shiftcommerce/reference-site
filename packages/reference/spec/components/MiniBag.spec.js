import { MiniBag } from '../../components/MiniBag'

test('renders the minibag links', () => {
  // arrange
  const cart = {
    lineItems: [
      {
        title: 'test',
        price: 10,
        discount: 0,
        quantity: 2,
        sku: '123',
        image_url: '',
        size: 'size - 8',
        productSku: '1231',
        productID: 1
      }
    ],
    totalQuantity: 0
  }
  const dispatch = () => {}

  // act
  const wrapper = shallow(
    <MiniBag cart={cart} dispatch={dispatch} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
})

test('renders the minibag links', () => {
  // arrange
  const cart = {
    lineItems: [],
    totalQuantity: 0
  }
  const dispatch = () => {}

  // act
  const wrapper = mount(
    <MiniBag cart={cart} dispatch={dispatch} />
  )

  // assert
  expect(wrapper.find('div.c-minibag__cart')).toIncludeText('0')
  expect(wrapper.find('div.c-minibag__cart-label--small')).toIncludeText('Basket')
  expect(wrapper.find('div.c-minibag__cart-label--large')).toIncludeText('View Your Bag')
})

test('renders the line item quantity as expected, where are items', () => {
  // arrange
  const cart = {
    lineItems: [
      {
        title: 'test',
        price: 10,
        discount: 0,
        quantity: 2,
        sku: '123',
        image_url: '',
        size: 'size - 8',
        productSku: '1231',
        productID: 1
      }
    ],
    totalQuantity: 2
  }
  const dispatch = () => {}

  // act
  const wrapper = mount(
    <MiniBag cart={cart} dispatch={dispatch} />
  )

  // assert
  expect(wrapper.find('div.c-minibag__cart')).toIncludeText('2')
  expect(wrapper.find('div.c-minibag__cart-label--small')).toIncludeText('Basket')
  expect(wrapper.find('div.c-minibag__cart-label--large')).toIncludeText('View Your Bag')
})

test('renders an active checkout button where cart has items', () => {
  // arrange
  const cart = {
    lineItems: [
      {
        title: 'test',
        price: 10,
        discount: 0,
        quantity: 2,
        sku: '123',
        image_url: '',
        size: 'size - 8',
        productSku: '1231',
        productID: 1
      }
    ],
    totalQuantity: 2
  }
  const dispatch = () => {}

  // act
  const wrapper = mount(
    <MiniBag cart={cart} dispatch={dispatch} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('CHECKOUT')
  expect(wrapper.find('.o-button')).not.toHaveClassName('o-button--disabled')
})

test('renders a disabled checkout button where cart is empty', () => {
  // arrange
  const cart = {
    lineItems: [],
    totalQuantity: 0
  }
  const dispatch = () => {}

  // act
  const wrapper = mount(
    <MiniBag cart={cart} dispatch={dispatch} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('CHECKOUT')
  expect(wrapper.find('.o-button')).toHaveClassName('o-button--disabled')
})
