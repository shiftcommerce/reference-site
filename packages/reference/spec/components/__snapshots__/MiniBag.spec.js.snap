// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a disabled checkout button where cart is empty 1`] = `
<MiniBag
  cart={
    Object {
      "lineItems": Array [],
      "totalQuantity": 0,
    }
  }
  dispatch={[Function]}
>
  <div>
    <Link
      href="/cart"
    >
      <a
        aria-label="View your cart"
        href="/cart"
        onClick={[Function]}
      >
        View Your Bag
        <span
          style={
            Object {
              "marginLeft": "5px",
              "marginRight": "5px",
            }
          }
        >
          (
          0
          )
        </span>
      </a>
    </Link>
    <Link
      href=""
    >
      <a
        aria-label="Go to checkout"
        className="o-button o-button--primary o-button--lrg o-button--disabled"
        href=""
        onClick={[Function]}
      >
        CHECKOUT
      </a>
    </Link>
  </div>
</MiniBag>
`;

exports[`renders an active checkout button where cart has items 1`] = `
<MiniBag
  cart={
    Object {
      "lineItems": Array [
        Object {
          "discount": 0,
          "image_url": "",
          "price": 10,
          "productID": 1,
          "productSku": "1231",
          "quantity": 2,
          "size": "size - 8",
          "sku": "123",
          "title": "test",
        },
      ],
      "totalQuantity": 2,
    }
  }
  dispatch={[Function]}
>
  <div>
    <Link
      href="/cart"
    >
      <a
        aria-label="View your cart"
        href="/cart"
        onClick={[Function]}
      >
        View Your Bag
        <span
          style={
            Object {
              "marginLeft": "5px",
              "marginRight": "5px",
            }
          }
        >
          (
          2
          )
        </span>
      </a>
    </Link>
    <Link
      href="/checkout"
    >
      <a
        aria-label="Go to checkout"
        className="o-button o-button--primary o-button--lrg"
        href="/checkout"
        onClick={[Function]}
      >
        CHECKOUT
      </a>
    </Link>
  </div>
</MiniBag>
`;

exports[`renders the minibag links 1`] = `
<div>
  <Link
    href="/cart"
  >
    <a
      aria-label="View your cart"
    >
      View Your Bag
      <span
        style={
          Object {
            "marginLeft": "5px",
            "marginRight": "5px",
          }
        }
      >
        (
        0
        )
      </span>
    </a>
  </Link>
  <Link
    href=""
  >
    <a
      aria-label="Go to checkout"
      className="o-button o-button--primary o-button--lrg o-button--disabled"
    >
      CHECKOUT
    </a>
  </Link>
</div>
`;
